src/Searchable/AggregatorObserver.php

```diff
@@ -85,12 +85,14 @@ public function saved($model): void
      */
     public function deleted($model): void
     {
-        if (static::syncingDisabledFor($model)) {
+        if (!$model->wasSearchableBeforeDelete()) {
             return;
         }
 
-        if ($this->usesSoftDelete($model) && config('scout.soft_delete', false)) {
-            $this->saved($model);
+        if ($this->usingSoftDeletes && $this->usesSoftDelete($model)) {
```

Comment:
Let's use the `config` method to ensure we have the latest value here; if the value would change between the construction of this class, and the method call to `deleted`, we would use the old value rather than the new one.
```suggestion
        if (config('scout.soft_delete', false) && $this->usesSoftDelete($model)) {
```